source_gen/lib/builder.dart

```diff
@@ -38,12 +38,20 @@ Builder combiningBuilder([BuilderOptions options]) {
 ///
 /// This will glob all files of the form `.*.g.part`.
 class CombiningBuilder implements Builder {
+  final bool _includePartName;
+
   @override
   Map<String, List<String>> get buildExtensions => const {
         '.dart': const [_outputExtensions]
       };
 
-  const CombiningBuilder();
+  /// Returns a new [CombiningBuilder].
+  ///
+  /// If [includePartName] is `true`, the name of each source part file
+  /// is output as a comment before its content. This can be useful when
+  /// debugging build issues.
+  const CombiningBuilder({bool includePartName = false})
```

Comment:
how would we use this? It's not plumbed through the options

Reply:
Oops...ack...

Reply:
PTAL